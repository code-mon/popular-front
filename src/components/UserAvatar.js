import React from 'react';

const containerStyle = {
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  padding: '0 0.5rem 0 0'
};

const imgStyle = {
  borderRadius: '50%',
  height: '3rem',
  marginRight: '0.25rem'
};

const UserAvatar = props => {
  const { user } = props;
  return (
    <div style={containerStyle}>
      <img style={imgStyle} src={user.w3.Paa} alt="avatar" />
      <p>{user.w3.ofa}</p>
    </div>
  );
};

export default UserAvatar;
