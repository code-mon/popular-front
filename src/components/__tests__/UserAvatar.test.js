import React from 'react';
import renderer from 'react-test-renderer';
import UserAvatar from '../UserAvatar.js';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

const user = {
  w3: {
    Paa:
      'https://lh6.googleusercontent.com/-daJSSSOVbi4/AAAAAAAAAAI/AAAAAAAADt0/FffcUzgmiwo/s96-c/photo.jpg',
    ofa: 'Jonathan'
  }
};

it('renders correctly', () => {
  const tree = renderer.create(<UserAvatar user={user} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('shows the users name', () => {
  const wrapper = shallow(<UserAvatar user={user} />);
  expect(wrapper.text()).toBe('Jonathan');
});
