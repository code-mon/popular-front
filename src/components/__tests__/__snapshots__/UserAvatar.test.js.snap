// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "space-between",
      "padding": "0 0.5rem 0 0",
    }
  }
>
  <img
    alt="avatar"
    src="https://lh6.googleusercontent.com/-daJSSSOVbi4/AAAAAAAAAAI/AAAAAAAADt0/FffcUzgmiwo/s96-c/photo.jpg"
    style={
      Object {
        "borderRadius": "50%",
        "height": "3rem",
        "marginRight": "0.25rem",
      }
    }
  />
  <p>
    Jonathan
  </p>
</div>
`;
